dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java

```diff
@@ -0,0 +1,721 @@
+            }
+        }
+
+        luceneQuery.append(
+                    browserQuery.showArchived
+                        ? " +(working:true OR deleted:true) "
+                        : browserQuery.showWorking
+                            ? " +working:true -deleted:true"
+                            : " +live:true");
+        return luceneQuery.toString();
+    }
+
+    private void includeLinks(final BrowserQuery browserQuery, final List<Map<String, Object>> returnList,
+                              final Role[] roles, final Folder parent, final Host host) throws DotDataException, DotSecurityException {
+        // Getting the links directly under the parent folder or host
+        List<Link> links = new ArrayList<>();
+
+        if (parent != null) {
+
+            if (browserQuery.showWorking) {
+
+                links.addAll(folderAPI.getLinks(parent, true, false, browserQuery.user, false));
+            } else {
+
+                links.addAll(folderAPI.getLiveLinks(parent, browserQuery.user, false));
+            }
+
+            if (browserQuery.showArchived) {
+
+                links.addAll(folderAPI.getLinks(parent, true, browserQuery.showArchived, browserQuery.user, false));
+            }
+        } else {
+
+            links = folderAPI.getLinks(host, true, browserQuery.showArchived, browserQuery.user, false);
+        }
+
+        for (final Link link : links) {
+
+            final List<Integer> permissions2 = permissionAPI.getPermissionIdsFromRoles(link, roles, browserQuery.user);
+
+            if (permissions2.contains(PERMISSION_READ)) {
+
+                final Map<String, Object> linkMap = link.getMap();
+                linkMap.put("permissions", permissions2);
+                linkMap.put("mimeType", "application/dotlink");
+                linkMap.put("name", link.getTitle());
+                linkMap.put("title", link.getName());
+                linkMap.put("description", link.getFriendlyName());
+                linkMap.put("extension", "link");
+                linkMap.put("hasLiveVersion", APILocator.getVersionableAPI().hasLiveVersion(link));
+                linkMap.put("statusIcons", UtilHTML.getStatusIcons(link));
+                linkMap.put("hasTitleImage", "");
+                linkMap.put("__icon__", "linkIcon");
+                returnList.add(linkMap);
+            }
+        }
+    } // includeLinks.
+
+    private void includeFolders(final BrowserQuery browserQuery,
+                                final List<Map<String, Object>> returnList,
+                                final Role[] roles,
+                                final Folder parent) throws DotDataException, DotSecurityException {
+
+        if (parent != null) {
+
+            List<Folder> folders = Collections.emptyList();
+            try {
+
+                folders = folderAPI.findSubFolders(parent, userAPI.getSystemUser(), false);
+            } catch (Exception e1) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)